mempool/src/tests/integration_tests.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::tests::test_framework::{
+    respond_to_request, send_message, test_transactions, MempoolTestFramework,
+};
+use diem_config::{config::PeerRole, network_id::NetworkId};
+use diem_types::PeerId;
+use futures::executor::block_on;
+use netcore::transport::ConnectionOrigin;
+use network::{
+    protocols::wire::handshake::v1::ProtocolIdSet,
+    testutils::{
+        builder::TestFrameworkBuilder,
+        test_framework::TestFramework,
+        test_node::{send_next_network_msg, NodeId, TestNode},
+    },
+    transport::ConnectionMetadata,
+    ProtocolId,
+};
+
+#[test]
+fn single_node_test() {
+    let mut test_framework: MempoolTestFramework = TestFrameworkBuilder::new()
```

Comment:
nit: Maybe this should read:

`... = TestFrameworkBuilder::new().add_num_nodes(1).create_validator(0).build();`?

Reply:
So, it's the number of unique owners, not the number of unique nodes, since the uniquness constraint is on the `owner index + node type`.  It makes most sense in a full network mock.  Though, the owners restriction is really not important.